packages/shared/dao/index.ts

```diff
@@ -204,73 +190,42 @@ export async function fetchCatalystStatuses(nodes: { domain: string }[]) {
         return {
           ...buildBaseCandidate(),
           layer,
-          type: 'layer-based',
-          score: layerScore(layer)
+          type: 'layer-based'
         }
       }
 
-      function buildIslandsCandidate(usersCount: number): IslandsBasedCandidate {
+      function buildIslandsCandidate(usersCount: number, usersParcels: Parcel[] | undefined, maxUsers: number | undefined): IslandsBasedCandidate {
         return {
           ...buildBaseCandidate(),
           usersCount,
-          type: 'islands-based',
-          score: islandsScore(usersCount)
+          maxUsers,
+          usersParcels,
+          type: 'islands-based'
         }
       }
 
       if (status === ServerConnectionStatus.OK) {
         if (result!.layers) {
           return [...union, ...result!.layers.map((layer) => buildLayerCandidate(layer))]
         } else {
-          return [...union, buildIslandsCandidate(result!.usersCount!)]
+          return [...union, buildIslandsCandidate(result!.usersCount!, result!.usersParcels, result!.maxUsers)]
         }
       } else return union
     },
     new Array<Candidate>()
   )
 }
 
-function candidateUsers(candidate: Candidate) {
-  return candidate.type === 'layer-based' ? candidate.layer.usersCount : candidate.usersCount
-}
-
-export function pickCatalystRealm(candidates: Candidate[]): Realm {
-  const usersByDomain: Record<string, number> = {}
-
-  candidates.forEach((it) => {
-    if (!usersByDomain[it.domain]) {
-      usersByDomain[it.domain] = 0
-    }
-
-    usersByDomain[it.domain] += candidateUsers(it)
-  })
+export function pickCatalystRealm(candidates: Candidate[], currentUserParcel: Parcel): Realm {
+  let config = getPickRealmsAlgorithmConfig(store.getState())
 
-  const sorted = candidates
-    .filter((it) => it)
-    .filter(
-      (it) =>
-        it.status === ServerConnectionStatus.OK &&
-        (it.type === 'islands-based' || it.layer.usersCount < it.layer.maxUsers)
-    )
-    .sort((c1, c2) => {
-      const elapsedDiff = c1.elapsed - c2.elapsed
-      const usersDiff = usersByDomain[c1.domain] - usersByDomain[c2.domain]
-      const scoreDiff = c2.score - c1.score
-
-      return Math.abs(elapsedDiff) > 1500
-        ? elapsedDiff // If the latency difference is greater than 1500, we consider that as the main factor
-        : scoreDiff !== 0
-        ? scoreDiff // If there's score difference, we consider that
-        : usersDiff !== 0
-        ? usersDiff // If the score is the same (as when they are empty)
-        : elapsedDiff // If the candidates have the same score by users, we consider the latency again
-    })
-
-  if (sorted.length === 0 && candidates.length > 0) {
-    throw new Error('No available realm found!')
+  if (!config || config.length === 0) {
+    config = defaultChainConfig
   }
 
-  return candidateToRealm(sorted[0])
+  const algorithm = createAlgorithm(config)
+
+  return candidateToRealm(algorithm.pickCandidate(candidates, currentUserParcel))
```

Comment:
Is the parcel always present?

Reply:
Yep. It is awaited here: https://github.com/decentraland/kernel/blob/c4437ba9c0423ff5fd57395cce8447b06ae9f667/packages/shared/dao/sagas.ts#L137